presto-main/src/main/java/io/prestosql/execution/scheduler/TopologyAwareNodeSelector.java

```diff
@@ -143,11 +140,11 @@ else if (!splitWaitingForAnyNode) {
             }
 
             InternalNode chosenNode = null;
-            int depth = networkLocationSegmentNames.size();
+            int depth = topologicalSplitCounters.size() - 1;
```

Comment:
This kind of `xxx - 1` calculation looks rare. Maybe we could get the same information from NetworkLocation? I'm also thinking is SegmentNames really needed.